### PR TITLE
Fix iOS, update example/ project, update freezed_anotation

### LINKED ISSUE #12: Problema con pods
Hola! tengo el siguiente problema y no me permite compilar para IOS. Actualicé Flutter a su ultima versión y recibo el siguiente error. ¿Alguien tiene alguna solución? Gracias de antemano

 [!] CocoaPods could not find compatible versions for pod "mercado_pago_mobile_checkout":
      In Podfile:
        mercado_pago_mobile_checkout (from `.symlinks/plugins/mercado_pago_mobile_checkout/ios`)

    Specs satisfying the `mercado_pago_mobile_checkout (from `.symlinks/plugins/mercado_pago_mobile_checkout/ios`)` dependency were found, but they
    required a higher minimum deployment target.

### PR DESCRIPTION
**Pull request by lautaropaske**
- Fix #12 as suggested in px-ios#2386. 
- Also, update `example` project (too old, wouldn't work otherwise).
- Bump `freezed_anotation` version to make this dependency compatible with projects using `freezed: ^0.12.0`

I'm opening this as a draft pull request as this is not yet a complete solution. Although this makes the plugin usable in iOS, the UI is left with a yellow topbar which I don't know the cause of.

Screen Shot 2021-01-08 at 17 48 08

Kudos to @lucasrueda

### CODE REVIEW

**Comment by lautaropaske** [comment]
Did you encounter a similar problem with the yellow topbar? @Gazer

**Comment by Gazer** [comment]
> Did you encounter a similar problem with the yellow topbar when creating the plugin? @Gazer

Yes, I think that is because of how px-ios uses the navigation controller, but not 100%.

File: example/android/app/build.gradle [hunk 1]
```diff
@@ -38,12 +38,11 @@ android {
 
     defaultConfig {
         // TODO: Specify your own unique Application ID (https://developer.android.com/studio/build/application-id.html).
-        applicationId "ar.com.p39.mercado_pago_mobile_checkout_example"
-        minSdkVersion 21
-        targetSdkVersion 28
+        applicationId "com.example.example"
+        minSdkVersion 16
```
Comment (Gazer) [inline example/android/app/build.gradle:42]: I think 16 is too low for px-android, lets keep it in 21.

File: example/android/app/src/debug/AndroidManifest.xml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
-    package="ar.com.p39.mercado_pago_mobile_checkout_example">
+    package="com.example.example">
```
Comment (Gazer) [inline example/android/app/src/debug/AndroidManifest.xml:2]: Can we keep the original package name?

File: example/android/app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -1,13 +1,13 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
-    package="ar.com.p39.mercado_pago_mobile_checkout_example">
+    package="com.example.example">
```
Comment (Gazer) [inline example/android/app/src/main/AndroidManifest.xml:2]: same

File: example/android/app/build.gradle (hunk 1 above)
Comment (lautaropaske) [reply example/android/app/build.gradle:42]: Yeah, my bad. I didn't test this PR for Android as no changes should affect it. That made me ignore the `example`'s Android part.

File: example/android/app/src/debug/AndroidManifest.xml (hunk 2 above)
Comment (lautaropaske) [reply example/android/app/src/debug/AndroidManifest.xml:2]: Yes! My bad. Same as above

File: example/android/app/src/main/AndroidManifest.xml (hunk 3 above)
Comment (lautaropaske) [reply example/android/app/src/main/AndroidManifest.xml:2]: +1

**Comment by lautaropaske** [comment]
Although I'm far from ok with the bug mentioned above, it works. I think merging this is better overall for people considering `px-flutter`. I don't really have much time now for digging deeper but I'd love to keep an eye on how this evolves and maybe develop a solution later. We could add a TODO in `README.md` with a warning.

If you're okay with that, @Gazer i'll add an update for the `README.md` and we can merge this PR.

**Comment by Gazer** [comment]
Yes, that would work for me.

Thanks for the time.

**Comment by Gazer** [comment]
Aplique todos estos cambios a mano en la version 0.0.3, no pude mergearlo porque habia demasiadas cosas que no queria que cambien pero sirvio de base, gracias por esto!

**Pull request closed without merging**